meta-resin-common/recipes-support/resin-init/resin-init-flasher/resin-init-flasher

```diff
@@ -160,6 +160,21 @@ else
     inform "No need to flash second stage bootloader to a specific device."
 fi
 
+inform "Waiting for device registration to complete."
+# Wait for the device to register before copying config.json
+STARTTIME=$(date +%s)
+ENDTIME=$(date +%s)
+TIMEOUT=30
+while [ "$(systemctl is-active resin-device-register.service)" != "active" ]
+do
+    if [ $((ENDTIME - STARTTIME)) -le $TIMEOUT ]; then
+        sleep 1
+        ENDTIME=$(date +%s)
+    else
+        fail "Timeout while waiting for register to finish. No network?"
```

Comment:
should the flashing fail if no network is present?

Reply:
I don't know. If there is no network present you will get a slightly different config.json as the device will not be registered. I guess the question is, if we don't require registration here why do we do it at all?

Reply:
The reason is because the device will opportunistically report flashing progress (via resin-device-progress) from the flasher image if it has network